### PR TITLE
Current User support for Email - SMTP Connector

### PR DESCRIPTION
**Pull request by IceOnly**
The most SMTP servers can be configured to allow a user to send emails in behalf of another mail address.
I added an option to the Setup Wizard to select if a specific sender or the User Settings should be used to send an email.

### CODE REVIEW

**Comment by JesperSchulz** [comment]
Apologies for the slight delay in processing this PR. We hope to get to do a review soon!

**Comment by darjoo** [comment]
@IceOnly Hi, we're currently in the works to do separate the dotnet of SMTP into a separate app of its own. I will process this as soon that work is done.

**Comment by IceOnly** [comment]
@[darjoo](https://github.com/darjoo) is there any time frame for it?

**Comment by darjoo** [comment]
> @[darjoo](https://github.com/darjoo) is there any time frame for it?

Hi @IceOnly, as my team and I are busy with the Launch Event and Directions prep this week. However, this is at the top of my agenda for next week.

File: Apps/W1/Email - SMTP Connector/app/src/SMTPConnectorSenderType.Enum.al [hunk 1]
```diff
@@ -0,0 +1,7 @@
+enum 4512 "SMTP Connector Sender Type"
```
Comment (darjoo) [inline Apps/W1/Email - SMTP Connector/app/src/SMTPConnectorSenderType.Enum.al:1]: // ------------------------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.
// ------------------------------------------------------------------------------------------------

File: Apps/W1/Email - SMTP Connector/app/src/SMTPMessage.Codeunit.al [hunk 2]
```diff
@@ -95,7 +120,7 @@ codeunit 4514 "SMTP Message"
         Recipients: List of [Text];
     begin
         // Add From
-        AddFrom(SMTPAccount."Sender Name", SMTPAccount."Email Address");
+        AddFrom(SMTPAccount);
```
Comment (darjoo) [inline Apps/W1/Email - SMTP Connector/app/src/SMTPMessage.Codeunit.al:123]: There are some changes in #15762 that overlap with this PR. But you could move these to `SMTPConnectorImpl` codeunit in the `BuildSMTPMessage` function first, that'd be great. As this codeunit no longer exists.

**Review by darjoo — CHANGES_REQUESTED** [review]
Great, just minor changes.

**Review by darjoo — COMMENTED** [review]
Great, I'll start processing it internally but could you have one more addition. A test to validate that if we create a current user account, we will get the details we expect. 
Like the test `TestMultipleAccountsCanBeRegistered` in `SMTPConnectorTest` codeunit.

**Comment by pri-kise** [comment]
@darjoo is there anything that can be done to get this PR merged?

**Review by darjoo — APPROVED** [review]
Minor changes in main repo but no diff in usage

**Comment by JesperSchulz** [comment]
Availability update: We will publish a fix for this issue in the next update for release 20.

Build ID to track: 43309.

**Pull request merged**